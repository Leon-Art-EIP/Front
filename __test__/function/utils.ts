import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
