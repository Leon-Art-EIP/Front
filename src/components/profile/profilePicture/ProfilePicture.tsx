import { cn } from "../../../tools/cn";
import ModifiableImage from "../../single-art-page/ModifiableImage";

export interface IProfilePictureProps {
  src: string;
  width?: number;
  height?: number;
  className?: string;
  imageClassName?: string;
  onClick?: () => void;
  modifiable?: boolean;
  style?: React.CSSProperties;
}

export default function ProfilePicture(props: IProfilePictureProps): JSX.Element {
  if (props.modifiable) {
    return (
      <ModifiableImage
        className={cn("rounded-full border-4", props.className)}
        imageClassName={cn("rounded-full", props.imageClassName)}
        hoverClassName="rounded-full"
        alt="profilePicture"
        src={props.src}
        width={props.width}
        height={props.height}
        onClick={props.onClick}
      />
    );
  }

  return (
    <img
      className={cn("rounded-full border-4", props.imageClassName)}
      alt="profilePicture"
      src={props.src}
      width={props.width}
      height={props.height}
      onClick={props.onClick}
      style={props.style}
    />
  );
}
