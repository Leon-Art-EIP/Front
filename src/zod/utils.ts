export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
