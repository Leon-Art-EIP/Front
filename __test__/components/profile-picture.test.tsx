import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import leonart from "../../public/leonart_logo.png";
import ProfilePicture from "../../src/components/profile/profilePicture/ProfilePicture";

test("ProfilePicture snapshot", () => {
  testSnapshot(<ProfilePicture src={leonart} />);
});
