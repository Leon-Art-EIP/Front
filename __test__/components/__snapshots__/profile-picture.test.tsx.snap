// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfilePicture snapshot 1`] = `
<img
  alt="profilePicture"
  className="rounded-full border-4"
  src="/public/leonart_logo.png"
/>
`;
